File: php/src/BL/BLUsuario.php
<?php
/** Lógica de negocio de Usuarios. Equivalente a BL/BLUsuario.vb. */

require_once __DIR__ . '/../DA/DAUsuario.php';

class BLUsuario
{
    private DAUsuario $da;

    public function __construct()
    {
        $this->da = new DAUsuario();
    }

    public function ConsultarUs(): array
    {
        return $this->da->ConsultarUsuarios();
    }

    public function CargarUsuario(string $cod): array
    {
        return $this->da->CargarUsuario($cod);
    }

    public function UsuariosAsociados(string $ccod_empresa): array
    {
        return $this->da->UsuariosAsociados($ccod_empresa);
    }

    public function InsertarUsuario(BEUsuario $obj, BEUser $conex): array
    {
        return $this->da->InsertarUsuario($obj, $conex);
    }

    public function InsertarUsuarioAdmin(BEUsuario $obj): bool
    {
        return $this->da->InsertarUsuarioAdmin($obj);
    }

    public function EditarUsuarioAdmin(BEUsuario $obj): bool
    {
        return $this->da->EditarUsuarioAdmin($obj);
    }

    public function EditarUsuario(BEUsuario $obj, BEUser $conex): array
    {
        return $this->da->EditarUsuario($obj, $conex);
    }

    public function EliminarUsuarioAdmin(string $cod, BEUser $obj): bool
    {
        return $this->da->EliminarUsuarioAdmin($cod, $obj);
    }

    public function EliminarUsuario(string $usuario, string $ipServidor, string $nomServidor, BEUser $obj): bool
    {
        return $this->da->EliminarUsuario($usuario, $ipServidor, $nomServidor, $obj);
    }

    /** @return array{0:bool,1:string} */
    public function ValidarBDEmpresa(string $ccod_empresa): array
    {
        return $this->da->ValidarBDEmpresa($ccod_empresa);
    }
}

File: BL/BLUsuario.vb
Public Class BLUsuario
    Dim objDA As New DA.DAUsuario()

    Public Function ConsultarUs()
        Return objDA.ConsultarUsuarios()
    End Function
    Public Function CargarUsuario(cod As String)
        Return objDA.CargarUsuario(cod)
    End Function
    Public Function UsuariosAsociados(ccod_empresa As String)
        Return objDA.UsuariosAsociados(ccod_empresa)
    End Function
    Public Function InsertarUsuario(objBE As BE.BEUsuario, objConex As BE.BEUser)
        Return objDA.InsertarUsuario(objBE, objConex)
    End Function
    Public Function InsertarUsuarioAdmin(objBE As BE.BEUsuario)
        Return objDA.InsertarUsuarioAdmin(objBE)
    End Function
    Public Function EditarUsuarioAdmin(objBE As BE.BEUsuario)
        Return objDA.EditarUsuarioAdmin(objBE)
    End Function
    Public Function EditarUsuario(objBE As BE.BEUsuario, objConex As BE.BEUser)
        Return objDA.EditarUsuario(objBE, objConex)
    End Function
    Public Function EliminarUsuarioAdmin(cod As String, obj As BE.BEUser)
        Return objDA.EliminarUsuarioAdmin(cod, obj)
    End Function
    Public Function EliminarUsuario(usuario As String, ipServidor As String, nomServidor As String, obj As BE.BEUser)
        Return objDA.EliminarUsuario(usuario, ipServidor, nomServidor, obj)
    End Function

    ' --- NEW: Validate child BD exists before any operation ---
    Public Function ValidarBDEmpresa(ccod_empresa As String) As Object
        Return objDA.ValidarBDEmpresa(ccod_empresa)
    End Function

End Class